src/components/ForecastApp.js

```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+
+import { FETCH_FORECAST_REQUEST } from '../constants';
+
+import Header from './Header';
+import ForecastInfo from './ForecastInfo';
+import Spinner from './Spinner';
+
+import './ForecastApp.css';
+
+class ForecastApp extends React.Component {
+  componentDidMount() {
+    const { fetchForecast } = this.props;
+    fetchForecast();
+  }
+
+  render() {
+    const { isFetching } = this.props;
+
+    return (
+      <div className='ForecastApp'>
+        <Header />
+
+        { isFetching.inProgress
+          ? <Spinner />
+          : <ForecastInfo />
+        }
+      </div>
+    )
+  }
+};
+
+ForecastApp.propTypes = {
+  isFetching: PropTypes.shape({
+    inProgress: PropTypes.bool.isRequired
+  }),
+  fetchForecast: PropTypes.func.isRequired
+};
+
+const mapStateToProps = (state) => {
+  return {
+    isFetching: state.isFetching
+  }
+};
+
+const mapDispatchToProps = (dispatch) => {
+  return {
+    fetchForecast: () => dispatch({ type: FETCH_FORECAST_REQUEST })
+  }
+};
+
+ForecastApp = connect(mapStateToProps, mapDispatchToProps)(ForecastApp);
```

Comment:
Do not reassign `class` name! Like never!

Reply:
Ok, got it. I saw this in the egghead course and was wondering about it.

Is it ok to do it like this?

```
export default connect(mapStateToProps)(ForecastApp);
```

Reply:
Yes and no. It's community rule to have components as a simple representation of data and create so called `containers` which are responsible for providing props to components. For example:

```javascript